src/pyaro/timeseries/Filter.py

```diff
@@ -1184,3 +1184,162 @@ def filter_stations(self, stations: dict[str, Station]) -> dict[str, Station]:
         selected_names = names[mask]
 
         return {name: stations[name] for name in selected_names}
+
+
+@registered_filter
+class ValleyFloorRelativeAltitudeFilter(StationFilter):
+    EARTH_RADIUS = 6378137  # meters
+
+    # https://cfconventions.org/Data/cf-conventions/cf-conventions-1.11/cf-conventions.html#latitude-coordinate
+    UNITS_LAT = set(
+        [
+            "degrees_north",
+            "degree_north",
+            "degree_N",
+            "degrees_N",
+            "degreeN",
+            "degreesN",
+        ]
+    )
+
+    # https://cfconventions.org/Data/cf-conventions/cf-conventions-1.11/cf-conventions.html#longitude-coordinate
+    UNITS_LON = set(
+        ["degrees_east", "degree_east", "degree_E", "degrees_E", "degreeE", "degreesE"]
+    )
+
+    def __init__(
+        self,
+        topo_file: str | None = None,
+        *,
+        radius: float = 5000,
+        topo_var: str = "Band1",
+        lower: float | None = None,
+        upper: float | None = None,
+    ):
+        self._topo_file = topo_file
+        self._topo_var = topo_var
+        self._radius = radius
+        self._lower = lower
+        self._upper = upper
+        self._topography = None
+
+    @property
+    def topography(self):
+        if "cf_units" not in sys.modules:
+            raise ModuleNotFoundError(
+                "valleyfloorrelativealtitudefilter filter is missing required dependency 'cf-units'. Please install to use this filter."
+            )
+        if "xarray" not in sys.modules:
+            raise ModuleNotFoundError(
+                "valleyfloorrelativealtitudefilter filter is missing required dependency 'xarray'. Please install to use this filter."
+            )
+
+        if self._topography is None:
+            try:
+                with xr.open_dataset(self._topo_file) as topo:
+                    topo = topo.fillna(0)
+                    self._topography = topo
```

Comment:
By storing self._topography you ensure that the file can't be closed and/or the data remains in memory for much longer then required. I guess the topography data is quite large, so maybe not a good idea.